opencti-platform/opencti-graphql/src/domain/stixCoreObject.js

```diff
@@ -80,31 +81,46 @@ export const findById = async (context, user, stixCoreObjectId) => {
 
 export const batchInternalRels = async (context, user, elements, opts = {}) => {
   const relIds = elements.map(({ element, definition }) => element[definition.databaseName]).flat().filter((id) => isNotEmptyField(id));
-  const resolvedElements = await internalFindByIds(context, user, relIds, { toMap: true });
-  return elements.map(({ element, definition }) => {
+  // Get all rel resolutions with system user
+  // The visibility will be restricted in the data preparation
+  const resolvedElements = await internalFindByIds(context, SYSTEM_USER, relIds, { toMap: true });
+  return await Promise.all(elements.map(async ({ element, definition }) => {
     const relId = element[definition.databaseName];
     if (definition.multiple) {
-      const relElements = (relId ?? []).map((id) => {
+      const relElements = await Promise.all((relId ?? []).map(async (id) => {
         const resolve = resolvedElements[id];
+        // If resolution is empty the database is inconsistent, an error must be thrown
         if (isEmptyField(resolve)) {
           throw UnsupportedError('Invalid loading of batched elements', { ids: relId });
         }
-        return resolve;
-      });
+        // If user have correct access right, return the element
+        if (await isUserCanAccessStoreElement(context, user, resolve)) {
+          return resolve;
+        }
+        // If access is not possible, return a restricted entity
+        return { id: resolve.internal_id, name: 'Restricted', entity_type: resolve.entity_type };
```

Comment:
Would it be possible for a given entity_type to NOT have a `name` attribute, leading to an impossible object schema-wise ? an opinion for instance.

Reply:
I have the same issue in my PR https://github.com/OpenCTI-Platform/opencti/pull/6323

Reply:
As discussed, if you're confident it concerns only entities with names, let's go for it as it would not be disruptive for the frontend side.